cli/cmd/inject.go

```diff
@@ -259,32 +183,79 @@ func InjectYAML(in io.Reader, out io.Writer) error {
 			return err
 		}
 
+		// The Kuberentes API is versioned and each version has an API modeled
+		// with its own distinct Go types. If we tell `yaml.Unmarshal()` which
+		// version we support then it will provide a representation of that
+		// object using the given type if possible. However, it only allows us
+		// to supply one object (of one type), so first we have to determine
+		// what kind of object `bytes` represents so we can pass an object of
+		// the correct type to `yaml.Unmarshal()`.
+
 		// Unmarshal the object enough to read the Kind field
 		var meta metaV1.TypeMeta
 		if err := yaml.Unmarshal(bytes, &meta); err != nil {
 			return err
 		}
 
-		var injected interface{} = nil
+		// obj and podTemplateSpec will reference zero or one the following
+		// objects, depending on the type.
+		var obj interface{} = nil
+		var podTemplateSpec *v1.PodTemplateSpec = nil
```

Comment:
Super minor -- `nil` is the zero-value for both of these types, so setting them to `nil` explicitly isn't necessary. You can write this as:

```
		var obj interface{}
		var podTemplateSpec *v1.PodTemplateSpec
```